### PR TITLE
TOOLS-2291 Implement txn oplog meta parser

### PR DESCRIPTION
**Pull request by xdg**
This PR implements just the oplog transaction entry parsing for metadata extraction.

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -92,14 +92,17 @@ type ApplyOpsResponse struct {
...
+	HistoryID  int64               `bson:"h"`
+	Version    int                 `bson:"v"`
+	Operation  string              `bson:"op"`
+	Namespace  string              `bson:"ns"`
+	Object     bson.D              `bson:"o"`
+	Query      bson.D              `bson:"o1,omitempty"`
+	UI         *primitive.Binary   `bson:"ui,omitempty"`
+	LSID       bson.Raw            `bson:"lsid,omitempty"`
+	TxnNumber  int64               `bson:"txnNumber,omitempty"`
+	PrevOpTime bson.Raw            `bson:"prevOpTime,omitempty"`
```
Comment (kmahar) [inline db/db.go]: if I understand correctly, we just parse this field so we can tell whether it was actually set. if it is set, that means this is part of a multi op transaction, and it is not the first entry in the transaction. 
is that true?

File: txn/meta.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	commit     bool
+	abort      bool
+	partial    bool
+	prepare    bool
+	prevOpTime string
+}
+
+// NewMeta extracts transaction metadata from an oplog entry.  A non-transaction will
+// return a zero-value Meta struct, not an error.
+func NewMeta(op db.Oplog) (Meta, error) {
```
Comment (kmahar) [inline txn/meta.go]: I don't see any cases where this method actually errors. do you see that changing in the future?

File: db/db.go (hunk 1 above)
Comment (xdg) [reply db/db.go]: Close.  It's always present, but for the first (or only) oplog entry of a transaction it is the "empty" value which is defined to mean timestamp of 0,0 and t of -1.

File: txn/meta.go (hunk 2 above)
Comment (xdg) [reply txn/meta.go]: Not currently, but in the future I'd like to change `Object` to be `bson.Raw` so it's not unmarshaling the entire, deep data structure of documents within the applyOps.  Then working with a raw value could potentially error, even if unlikely.  I'll add a comment about it.

**Pull request closed without merging**